git_machete/client.py

```diff
@@ -1683,13 +1683,15 @@ def checkout_github_prs(self, pr_no: int) -> None:
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
 
-        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
-        pr = utils.find_or_none(lambda pr: pr.number == pr_no, all_prs)
-
+        pr = get_pull_request_by_number(str(pr_no), org, repo)
         if not pr:
             raise MacheteException(f"PR #{pr_no} is not found in repository `{org}/{repo}`")
+
+        if pr.state == 'closed':
+            warn(f'Pull request #{pr_no} is already closed.')
         debug('checkout_github_pr()', f'found {pr}')
 
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
```

Comment:
Do `all_prs` include closed PRs as well? 🤔

Reply:
No. Initially i wanted to grab all PRs here, but in case of git-machete repository it extended execution time for about 0.2-0.3s. In case of bigger/older repositories execution time would much longer, so i dropped this idea. Instead i am asking API for certain PR number in `get_pull_request_by_number_or_none`.

Reply:
Plus there would changes required in `anno-prs` if `all_prs` holds not only open PRs.

Reply:
Ok! pls rename this var accordingly then to `all_open_prs`